File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        clean(s);
        int n=s.size();
        int l=0, r=n-1;
        while (l < r){
            if(s[l] != s[r]) return false; 
            l++; 
            r--;
        }
        return true;
    }

    void clean(std::string& s) {
    size_t writeIdx = 0;
    for (size_t readIdx = 0; readIdx < s.length(); ++readIdx) {
        unsigned char c = s[readIdx];
        if (std::isalnum(c)) {
            s[writeIdx++] = std::tolower(c);
        }
    }
    s.resize(writeIdx);
    }
};

File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        ss = "".join(char for char in s if char.isalnum())
        ss = ss.lower()
        l, r = 0, len(ss)-1
        while(l < r):
            if ss[l] != ss[r]:
                return False
            l+=1
            r-=1
        return True